pkg/gzip/gzip.go

```diff
@@ -165,3 +191,11 @@ func ExtractTarGz(owner string, fp io.Reader, dest string) (err error) {
 	}
 	return nil
 }
+
+func firstExistingParent(path string) string {
+	for _, err := os.Stat(path); os.IsNotExist(err); {
+		path = filepath.Dir(path)
+		_, err = os.Stat(path)
```

Comment:
If I'm following things, as it's written this code does the Stat operation twice for the same value of `path`. I think the intent of this code would be clearer if it were written as a for expression like you suggest.